internal/app/machined/pkg/controllers/network/route_spec.go

```diff
@@ -203,7 +203,15 @@ func (ctrl *RouteSpecController) syncRoute(ctx context.Context, r controller.Run
 				existing.Protocol == uint8(route.TypedSpec().Protocol) &&
 				existing.Attributes.OutIface == linkIndex && existing.Attributes.Priority == route.TypedSpec().Priority &&
 				(value.IsZero(route.TypedSpec().Source) ||
-					existing.Attributes.Src.Equal(route.TypedSpec().Source.AsSlice())) {
+					existing.Attributes.Src.Equal(route.TypedSpec().Source.AsSlice())) &&
+				(func(in uint32, existing *rtnetlink.RouteMessage) bool {
+					if existing.Attributes.Metrics != nil {
+						return in == existing.Attributes.Metrics.MTU
+					}
+
+					// zero is default value
+					return in == 0
+				}(route.TypedSpec().MTU, existing)) {
```

Comment:
for debugging purposes we might add `old_mtu`/`new_mtu` to the `logger.Debug` at line 217/225